structuralcodes/sections/_rc_utils.py

```diff
@@ -58,27 +74,41 @@ def create_surface_geometries(polygons_list, material):
     z_na = -eps / curv  # distance to neutral fibre
 
     # Cutting concrete geometries and retaining the compressed block
-    cut_geom = CompoundGeometry(None, None)
-    for i, part in enumerate(rotated_geometry.geometries):
-        min_x, max_x, min_y, max_y = part.calculate_extents()
-        upper_div, lower_div = part.split(((min_x, z_na), 0))
+    cut_geom = None
+    for part in rotated_geometry.geometries:
+        upper_div, lower_div = part.split(((0, z_na), 0))
         # Convert to SurfaceGeometry
-        subpart_sg = create_surface_geometries(upper_div, part.material)
-        if i == 0:
-            cut_geom = CompoundGeometry(subpart_sg)
+        subpart = create_surface_geometries(upper_div, part.material)
+        if cut_geom is None:
+            cut_geom = subpart
         else:
-            cut_geom += subpart_sg
+            cut_geom += subpart
 
     # Add reinforcement geometries
     for reinf in rotated_geometry.point_geometries:
         cut_geom += reinf
 
     # return geoemtry to original rotation
-    cracked_geom = cut_geom.rotate(-theta)
+    cracked_geom = cut_geom.rotate(theta)
 
     # Define the cracked section
-    cracked_sec = GenericSection(cracked_geom)
+    if isinstance(section.section_calculator.integrator, FiberIntegrator):
+        mesh_size = getattr(
+            section.section_calculator.integrator, 'mesh_size', 0.01
+        )
+        cracked_sec = GenericSection(
+            cracked_geom,
+            integrator='fiber',
+            mesh_size=mesh_size,
+        )
+    else:
+        cracked_sec = GenericSection(cracked_geom, integrator='marin')
```

Comment:
not your fault here, you had to write it like this because the API of the section is a little bit lacking methods for doing this kind of operations so you had to do in this intricate way... Probably I will improve the API after merging this...

Reply:
Good point. I have created a new issue (#211) for this and assigned it to you @talledodiego.